### PR TITLE
2384 - Add build verification tool

### LINKED ISSUE #2384: Build: Make a test to verify the dist
**Describe the bug**
We found that some things are missing from our dist (backwards compatible min files and svg-extended). We wanted some tests for this. Not 100% sure how that would work but something like

**To Verify**
1. Run the build command
2. Check each file is as expectedly copied into dist
3. Test pass or fail

**Other thoughts**
Do we verify cdn links? https://cdn.hookandloop.infor.com/sohoxi/4.17.0/js/sohoxi.min.js
Do we verify npm download?

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR adds a tool to the build process to ensure that all files that need to be generated at build time are present in the `/dist` folder.  This script has been added to the end of the NPM `build:release` task.

**Related github/jira issue (required)**:
Closes #2384 

**Steps necessary to review your pull request (required)**:
Pull this branch and run `npm run build:release`.  At the end of the build, the output should be similar to:
```sh
ⓘ  Verifying the last build...
    folders: 106
    files: 264
🍺  All (264) expected files have been found!
```
Next, go into the `/dist` folder and delete any number files.  Then, run the script directly with `node ./scripts/verify.js`.  The output should be similar to:
```sh
ⓘ  Verifying the last build...
    folders: 106
    files: 260
ⓧ  (4) expected files are missing from the last build:
    /dist/css/theme-soho-contrast.min.css
    /dist/js/sohoxi.js.map
    /dist/sass/src/layouts/_object-summary.scss
    /dist/svg/theme-soho-svg.html
```
It's also possible to generate a new list of expected files based off of the current state of the `/dist` folder. To test this process, run `node ./scripts/verify.js --rebuild` against the deletions just made.  The output should be similar to:
```sh
ⓘ  Creating a new expected-files.json list...
   folders: 106
   files: 260
🍺  New file list saved to "/Users/ecoyle/dev/project-hardcore/scripts/data/expected-files.json"
```
Also check that the NPM scripts command for the verify script works by running:
```sh
npm run build:verify
```

**Included in this Pull Request**:
- A note to the change log.

### CODE REVIEW

File: scripts/logger.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const INFO_I = '\u24D8';
 const WARN = '\u26A0';
 const NDASH = '\u2013';
 const CALLOUT = '\u00BB';
-const BEER = '\u1F37A';
+const BEER = '🍺';
```
Comment (clepore) [inline scripts/logger.js]: 🍻

File: scripts/verify.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+// Do a file glob against `/dist`.
+glob(`${paths.dist}/**/*`, globOptions, (err, files) => {
+  if (err) {
+    logger('error', err);
+    return;
+  }
+
+  // TODO:
+  // - Count against a predetermined list
+  // - Pass/Fail properly with Process Exit Codes
```
Comment (clepore) [inline scripts/verify.js]: Do we need these "todo" comments anymore?

File: package.json [hunk 3]
```diff
@@ -63,7 +63,8 @@
     "release:beta": "release-it --preRelease=beta --no-github.release",
     "release:rc": "release-it --preRelease=rc --no-github.release",
     "release:final": "release-it",
-    "test": "npm run mdlint && npm run eslint && npm run stylelint && npm run functional:ci && npm run e2e:ci"
+    "test": "npm run mdlint && npm run eslint && npm run stylelint && npm run functional:ci && npm run e2e:ci",
+    "verify": "node ./scripts/verify.js"
```
Comment (tmcconechy) [inline package.json]: Maybe `build:verify` actually?

File: scripts/data/expected-files.json [hunk 4]
```diff
@@ -0,0 +1,266 @@
+[
+	"/dist/css/dark-theme.css",
+	"/dist/css/dark-theme.css.map",
+	"/dist/css/dark-theme.min.css",
```
Comment (tmcconechy) [inline scripts/data/expected-files.json]: Shouldnt we have a map for both the full css and the min.css?

File: scripts/verify.js (hunk 2 above)
Comment (EdwardCoyle) [reply scripts/verify.js]: Noooope

File: scripts/data/expected-files.json (hunk 4 above)
Comment (EdwardCoyle) [reply scripts/data/expected-files.json]: No need to add a map file for the minified css, since they are production-ready and don't need to be debugged.

**Pull request merged**